dl.go

```diff
@@ -36,12 +36,16 @@ func (m *MagnetTarget) DownloadFuture(ctx context.Context, url string) func() (u
 		errC <- err
 	}()
 
-	return func() (string, string, error) {
-		err := <-errC
-		if err != nil {
-			return url, "", trace.Wrap(err)
+	return func(ctx context.Context) (string, string, error) {
+		select {
+		case err := <-errC:
+			if err != nil {
+				return url, "", trace.Wrap(err)
+			}
+			return url, path, nil
+		case <-ctx.Done():
```

Comment:
The DownlodFuture takes a context itself and passes that to the underlying download function, does that not release properly if the context is cancelled? Otherwise it seems redundant that both the base function and the future each take a context that can cancel the download.

Reply:
Yes, you're right. I should've thought about that as well.